lightning/src/routing/router.rs

```diff
@@ -30,47 +30,60 @@ use crate::prelude::*;
 use crate::sync::Mutex;
 use alloc::collections::BinaryHeap;
 use core::cmp;
-use core::ops::Deref;
+use core::ops::{Deref, DerefMut};
 
 /// A [`Router`] implemented using [`find_route`].
 pub struct DefaultRouter<G: Deref<Target = NetworkGraph<L>>, L: Deref, S: Deref> where
 	L::Target: Logger,
 	S::Target: for <'a> LockableScore<'a>,
+	S::Target: Score,
 {
 	network_graph: G,
 	logger: L,
 	random_seed_bytes: Mutex<[u8; 32]>,
-	scorer: S
+	scorer: S,
+	score_params: <S::Target as Score>::ScoreParams
 }
 
 impl<G: Deref<Target = NetworkGraph<L>>, L: Deref, S: Deref> DefaultRouter<G, L, S> where
 	L::Target: Logger,
 	S::Target: for <'a> LockableScore<'a>,
+	S::Target: Score,
 {
 	/// Creates a new router.
-	pub fn new(network_graph: G, logger: L, random_seed_bytes: [u8; 32], scorer: S) -> Self {
+	pub fn new(
+		network_graph: G,
+		logger: L,
+		random_seed_bytes: [u8; 32],
+		scorer: S,
+		score_params: <S::Target as Score>::ScoreParams
+	) -> Self {
 		let random_seed_bytes = Mutex::new(random_seed_bytes);
-		Self { network_graph, logger, random_seed_bytes, scorer }
+		Self { network_graph, logger, random_seed_bytes, scorer, score_params }
 	}
 }
 
-impl<G: Deref<Target = NetworkGraph<L>>, L: Deref, S: Deref> Router for DefaultRouter<G, L, S> where
+impl< G: Deref<Target = NetworkGraph<L>>, L: Deref, S: Deref> Router for DefaultRouter< G, L, S> where
 	L::Target: Logger,
 	S::Target: for <'a> LockableScore<'a>,
```

Comment:
the problem I have is generally with this part `for<'a> LockableScore<'a>` 
I seem to not be able to properly set the types or annotate the lifetimes for the score params for this type bounds in the `DefaultRouter` object

I've tried removing the `for <'a>` and explicitly passing in the lifetimes and using `'a` in setting the type of score_params in `DefaultRouter` to be `<<<S::Target as LockableScore<'a>::Locked> as Score>::ScoreParams` but from there I get a complaint from the compiler that `'a` might outlive `&self`
I've also tried adding `where Self:'a` to that, notifying the compiler that `self` does outlive `'a`, but other problems surface, and I don't think it's going in the right direction. 

Overall I think this isn't really a lifetimes issue, but really how to represent `for <'a> LockableScore <'a>` in a form that is lifetime agnostic. and to set a type to cast into `Score` so that I can do a `Score::ScoreParams`

Reply:
>I've tried removing the for <'a> and explicitly passing in the lifetimes and using 'a in setting the type of score_params in DefaultRouter to be <<<S::Target as LockableScore<'a>::Locked> as Score>::ScoreParams but from there I get a complaint from the compiler that `'a` might outlive `&self`

<img width="1005" alt="image" src="https://user-images.githubusercontent.com/906931/235967515-04fd38d4-0ce3-4dec-a885-43275c9fe2b1.png">